Change/upgraded to react18 CMEM-6639

arausly opened the pull request:
Made refactors to adapt for version 18 and even version 19.

arausly committed:
adapted for react 18

arausly committed:
added temp any type for failing strict ts check

arausly committed:
merge develop

arausly committed:
corrected interface to fix compilation errors

arausly committed:
fixed compile errors in ContextOverlay

andreas-schultz committed:
Merge branch 'develop' into change/upgradedToReact18-CMEM-6639

# Conflicts:
#	src/cmem/ContentBlobToggler/StringPreviewContentBlobToggler.tsx

andreas-schultz committed:
Remove seemingly unnecessary changes

andreas-schultz committed:
Upgrade react testing library from v12 to v14

andreas-schultz committed:
Fix deprecation warnings because of React 18 upgrade

haschek committed:
add NotAvailable component

haschek committed:
add tooltip indicator if no tag is used

haschek committed:
provide an inline option for content blob togglers

haschek commented on src/cmem/react-flow/configuration/graph.ts:
```diff
@@ -16,7 +15,7 @@ const edgeTypes = {
     danger: EdgeBezier,
 };
 
-const nodeTypes: Record<GRAPH_NODE_TYPES, React.ReactNode /*& ComponentType<NodeProps>*/> = {
```
We need to check this change from the LTS version because this is used for both supported react flow versions.

haschek commented on src/cmem/react-flow/configuration/linking.ts:
```diff
@@ -14,7 +13,7 @@ const edgeTypes = {
     danger: EdgeStep,
 };
 
-const nodeTypes: Record<LINKING_NODE_TYPES, React.ReactNode /*& ComponentType<NodeProps>*/> = {
```
same here.

haschek commented on src/cmem/react-flow/configuration/workflow.ts:
```diff
@@ -12,7 +11,7 @@ const edgeTypes = {
     danger: EdgeStep,
 };
 
-const nodeTypes: Record<WORKFLOW_NODE_TYPES, React.ReactNode /*& ComponentType<NodeProps>*/> = {
```
same here.

haschek commented on src/components/Application/ApplicationSidebarNavigation.tsx:
```diff
@@ -4,14 +4,18 @@ import { SideNav as CarbonSideNav, SideNavProps as CarbonSideNavProps } from "@c
+            children: React.ReactNode;
+            className?: string;
+            onToggle?: any //todo change later
+        }
```
Looks like a hack, also `React.HTMLAttributes<HTMLElement>` was taken out. We need to check for the reasons.

haschek commented on src/components/Application/ApplicationTitle.tsx:
```diff
@@ -60,7 +60,13 @@ export const ApplicationTitle = ({
+                                : depiction instanceof HTMLElement
+                                    ? <>{depiction.outerHTML}</>
+                                    : depiction}
+                        </span>
```
Why should land a HTMLElement here?

andreas-schultz replied:
This is how the depiction parameter has been defined  years ago. With the React upgrade this turned into a type error. Can the `ImgDepiction` and `SvgDepiction` types be removed?

haschek commented on src/components/PropertyValuePair/PropertyValueList.tsx:
```diff
@@ -22,7 +22,7 @@ export const PropertyValueList = ({
               const originalChild = child as React.ReactElement;
               if (originalChild && originalChild.type && originalChild.type === PropertyValuePair) {
-                  return React.cloneElement(originalChild, { singleColumn: true });
+                  return React.cloneElement(originalChild as React.ReactElement<{singleColumn: boolean}>, { singleColumn: true });
```
Type i set earlier, why need this be overwritten here?

haschek commented on src/components/PropertyValuePair/PropertyValuePair.tsx:
```diff
@@ -37,7 +37,7 @@ export const PropertyValuePair = ({
               const originalChild = child as React.ReactElement;
               if (originalChild.type && (originalChild.type === PropertyName || originalChild.type === PropertyValue)) {
-                  return React.cloneElement(originalChild, { nowrap: true });
+                  return React.cloneElement(originalChild as React.ReactElement<{nowrap: boolean}>, { nowrap: true });
```
Same here.

haschek commented on src/extensions/react-flow/edges/EdgeLabel.tsx:
```diff
@@ -109,7 +109,7 @@ export interface EdgeLabelObjectProps extends React.SVGAttributes<SVGForeignObje
     ({ children, edgeCenter, resizeTimeout = -1, ...otherForeignObjectProps }: EdgeLabelObjectProps) => {
         const containerCallback = React.useCallback(
-            (containerRef) => {
+            (containerRef:any) => {
```
There is probably another type for refs.

haschek commented on src/extensions/react-flow/nodes/NodeContent.tsx:
```diff
@@ -735,7 +735,7 @@ export function NodeContent<CONTENT_PROPS = React.HTMLAttributes<HTMLElement>>({
     };
 
-    const onResize = React.useCallback((_0, _1, _2, d) => {
+    const onResize = React.useCallback((_0:any, _1:any, _2:any, d:any) => {
```
Should be probably the same like in line 759?

haschek committed:
add story for NotAvailable and update changelog

haschek committed:
update yarn lock

haschek committed:
Merge remote-tracking branch 'origin/change/upgradedToReact18-CMEM-6639' into feature/replaceDeprecatedGuiComponents-CMEM-4905

andreas-schultz committed:
Add isValidNewOption to MultiSuggestField

andreas-schultz committed:
Fix tests

haschek committed:
fix border of tag in multi suggest component

andreas-schultz committed:
Remove unnecessary code leading to test failure

haschek committed:
Merge remote-tracking branch 'origin/change/upgradedToReact18-CMEM-6639' into feature/replaceDeprecatedGuiComponents-CMEM-4905

haschek committed:
use var for header height

haschek committed:
simplify fetching styles for small/large text blobs

haschek committed:
add togglerSize property

haschek committed:
support the size property of the BlueprintJS button by forwarding it to the icon correctly

haschek committed:
use jest-fixed-jsdom to fix global JS APIs

haschek committed:
use jest-fixed-jsdom to fix global JS APIs

andreas-schultz committed:
Several MultiSelect improvements

    -   `searchListPredicate` property: Allows to filter the complete list of search options at once.
    -   Following optional BlueprintJs properties are forwarded now to override default behaviour: `noResults`, `createNewItemRenderer` and `itemRenderer`
    -   by default, if no searchPredicate or searchListPredicate is defined, the filtering is done via case-insensitive multi-word filtering.
    -   `searchPreficate`: replaced by the, in some cases, more efficient `searchListPredicate`

andreas-schultz committed:
Fix imports to fix tests

andreas-schultz committed:
Fix all imports in components to not import from index

haschek committed:
fix imports for storybook to prevent circular dpendencies

haschek committed:
fix more imports in stories

haschek committed:
fix imports

andreas-schultz committed:
Merge branch 'develop' into change/upgradedToReact18-CMEM-6639

# Conflicts:
#	src/extensions/codemirror/CodeMirror.tsx

andreas-schultz committed:
Fix new compile issues after merge from develop

andreas-schultz committed:
Merge branch 'change/upgradedToReact18-CMEM-6639' into feature/replaceDeprecatedGuiComponents-CMEM-4905

# Conflicts:
#	CHANGELOG.md
#	src/cmem/ActivityControl/ActivityControlWidget.tsx
#	src/components/MultiSelect/MultiSelect.tsx

andreas-schultz committed:
Refactor imports from index

andreas-schultz committed:
Remove potential hook re-ordering problem in VisualTour component

andreas-schultz committed:
Reset drop down list immediately in MultiSelect when query changes

haschek committed:
Merge branch 'develop' into change/upgradedToReact18-CMEM-6639

andreas-schultz committed:
Merge change/upgradedToReact18-CMEM-6639 into feature/replaceDeprecatedGuiComponents-CMEM-4905

andreas-schultz committed:
Fix re-rendering bug in MultiSelect

andreas-schultz committed:
Merge pull request #367 from eccenca/feature/replaceDeprecatedGuiComponents-CMEM-4905

Replace deprecated gui components (CMEM-4905)

andreas-schultz committed:
Remove uncommented code

andreas-schultz committed:
Remove type hack in ApplicationSidebarNavigation

andreas-schultz committed:
Improve types in various places

andreas-schultz committed:
Fix compile issue

haschek committed:
allow change/ refix for branch names to create feature/fix packages

haschek committed:
fix import

haschek committed:
document React version change

haschek committed:
Merge remote-tracking branch 'origin/develop' into change/upgradedToReact18-CMEM-6639

haschek committed:
fix changelog, some items were added to wrong sections

haschek committed:
update deprecated note, fix version

haschek committed:
remove extra react types resolution, should not be necessary atm
